frontend/ui/widget/dictquicklookup.lua

```diff
@@ -61,6 +61,26 @@ local highlight_strings = {
     unhighlight = _("Unhighlight"),
 }
 
+function DictQuickLookup:canSearch()
+    if self:isDocless() then
+        return false
+    end
+
+    if self.is_wiki then
+        -- In the Wiki variant of this widget, the Search button is coopted to cycle between enabled languages.
+        if #self.wiki_languages > 1 then
+            return true
+        end
+    else
+        -- This is to prevent an ineffective button when we're launched from the Reader's menu.
+        if self.ui.highlight.selected_text then
+            return true
+        end
```

Comment:
(Not an issue with out current code because of some side effects, but...)
This would probably be cleaner:
```diff
--- a/frontend/apps/reader/modules/readerhighlight.lua
+++ b/frontend/apps/reader/modules/readerhighlight.lua
@@ -395,4 +395,5 @@ function ReaderHighlight:clear(clear_id)
     if self.hold_pos then
         self.hold_pos = nil
+        self.selected_word = nil
         self.selected_text = nil
         UIManager:setDirty(self.dialog, "ui")
@@ -1195,5 +1196,4 @@ function ReaderHighlight:onHoldRelease()
     elseif self.selected_word then
         self:lookup(self.selected_word, self.selected_link)
-        self.selected_word = nil
     end
     return true
--- a/frontend/ui/widget/dictquicklookup.lua
+++ b/frontend/ui/widget/dictquicklookup.lua
@@ -76,5 +76,5 @@ function DictQuickLookup:canSearch()
     else
         -- This is to prevent an ineffective button when we're launched from the Reader's menu.
-        if self.ui.highlight.selected_text then
+        if self.ui.highlight.selected_word or self.ui.highlight.selected_text then
             return true
         end
```
Normally, `selected_text` is only set on multiple word selection - it is empty on single word long-press, where `selected_word` is set and used instead.

It happens that when DictQuickLookup is shown that the test for the unrelated <kbd>Highlight</kbd> button is updating selected_text:
```
frontend/document/credocument.lua:616: in function 'getTextFromPositions'
frontend/apps/reader/modules/readerhighlight.lua:1243: in function 'highlightFromHoldPos'
frontend/apps/reader/modules/readerhighlight.lua:1317: in function 'getHighlightedItem'
frontend/ui/widget/dictquicklookup.lua:926: in function 'getHighlightText'
frontend/ui/widget/dictquicklookup.lua:449: in function 'init'
```
So, this side effect makes it work.

It happens that I remove/replace this <kbd>Highlight</kbd> button in my own patch, so I got the <kbd>Search</kbd> button always disabled :) (Had a hard time finding this was the issue.)

Reply:
Or instead, just:
```diff
         -- This is to prevent an ineffective button when we're launched from the Reader's menu.
-        if self.ui.highlight.selected_text then
+        if self.highlight then
```
if we don't want to take much risks, the same check used for the <kbd>Highlight</kbd> button:
https://github.com/koreader/koreader/blob/9f835156d16da2f958944a7f2326ee1cc9b47af3/frontend/ui/widget/dictquicklookup.lua#L446-L449

Reply:
Yeah, I remember that some of these were only set in certain circumstances, but I don't remember the details, and I think I only went so far as thinking "depends on what did the selection" or something.

I'm not *quite* sure `self.highlight` was set in the specific set of circumstances this tried to fix though, but if it works, go ahead ;).